controllers/users.js

```diff
@@ -167,18 +167,45 @@ const getUsers = async (req, res) => {
     if (transformedQuery?.filterBy === OVERDUE_TASKS) {
       try {
         const tasksData = await getOverdueTasks(days);
-        const users = new Set();
+        const userIds = new Set();
+        const usersData = [];
+
         tasksData.forEach((task) => {
-          users.add(task.assignee);
+          userIds.add(task.assignee);
         });
+
+        for (const userId of Array.from(userIds)) {
+          const userInfo = await fetchUser({ userId: userId });
+
+          if (userInfo) {
+            const userTasks = tasksData.filter((task) => task.assignee === userId);
+            if (dev) {
+              const userData = {
+                id: userId,
+                discordId: userInfo.user.discordId,
+                username: userInfo.user.username,
+                tasks: userTasks,
+              };
+              usersData.push(userData);
+            } else {
+              const userData = {
+                id: userId,
+                discordId: userInfo.user.discordId,
+                username: userInfo.user.username,
+              };
+              usersData.push(userData);
+            }
+          }
+        }
+
         return res.json({
-          message: "Users returned successfully!",
-          count: users.size,
-          users: Array.from(users),
+          message: "Users and their tasks returned successfully!",
+          count: usersData.length,
+          users: usersData,
```

Comment:
Here if the feature flag is not enabled then still it returns this message which is inappropriate as tasks are not there

Reply:
Done